tests/beta/test_default_tensor.py

```diff
@@ -374,294 +414,420 @@ def circuit():
 
+        assert dev._contracted_state_node is None
+        dev._contract_premeasurement_network()
+        assert dev._contracted_state_node is not None
+        cont_state = dev._contracted_state_node
+        dev._contract_premeasurement_network()  # should not change anything
+        assert dev._contracted_state_node == cont_state
 
-        def test_add_initial_state_nodes_3_wires_biseparable_A_BC(self, tensornet_device_3_wires):
-            """Tests that biseparable A|BC initial states are properly created for a 3 wire device."""
+        expected = np.outer(
+            np.outer([0.0, 1.0], [1 / np.sqrt(2), 1 / np.sqrt(2)]), [1.0, 0.0]
+        ).reshape([2, 2, 2])
 
-        dev = tensornet_device_3_wires
-        dev._clear_network_data()
+        assert np.allclose(cont_state.tensor, expected, atol=tol, rtol=0)
+        assert cont_state.name == "Ket"
 
-        tensors = [np.array([[1., 0.], [0., 1.]]) / np.sqrt(2), np.array([1., 1.]) / np.sqrt(2)]
-        wires = [[0], [1, 2]]
-        names = ["AliceState", "BobCharlieState"]
-        dev._add_initial_state_nodes(tensors, wires, names)
+    @pytest.mark.parametrize("method", ["auto", "greedy", "branch", "optimal"])
+    def test_state(self, rep, method, tol):
+        """Tests that the _state method produces the correct state after contraction."""
 
-        assert "state" in dev._nodes and len(dev._nodes) == 1
-        assert len(dev._nodes["state"]) == 2
-        assert dev._nodes["state"][0].name == "AliceState(0,)"
-        assert dev._nodes["state"][1].name == "BobCharlieState(1, 2)"
-        assert edges_valid(dev, num_nodes=2)
+        dev = qml.device("default.tensor", wires=3, representation=rep)
 
+        dev.apply("PauliX", [0], [])
+        dev.apply("Hadamard", [1], [])
 
-        def test_add_initial_state_nodes_3_wires_biseparable_AC_B(self, tensornet_device_3_wires):
-            """Tests that biseparable AC|B initial states are properly created for a 3 wire device."""
+        assert dev._contracted_state_node is None
+        ket = dev._state()
+        assert dev._contracted_state_node is not None
 
-        dev = tensornet_device_3_wires
-        dev._clear_network_data()
+        expected = np.outer(
+            np.outer([0.0, 1.0], [1 / np.sqrt(2), 1 / np.sqrt(2)]), [1.0, 0.0]
+        ).reshape([2, 2, 2])
+        assert np.allclose(ket, expected, atol=tol, rtol=0)
 
-        tensors = [np.array([[1., 0.], [0., 1.]]) / np.sqrt(2), np.array([1., 1.]) / np.sqrt(2)]
-        wires = [[0, 2], [1]]
-        names = ["AliceCharlieState", "BobState"]
-        dev._add_initial_state_nodes(tensors, wires, names)
 
-        assert "state" in dev._nodes and len(dev._nodes) == 1
-        assert len(dev._nodes["state"]) == 2
-        assert dev._nodes["state"][0].name == "AliceCharlieState(0, 2)"
-        assert dev._nodes["state"][1].name == "BobState(1,)"
-        assert edges_valid(dev, num_nodes=2)
+class TestDefaultTensorNetworkMoreParametrize:
```

Comment:
These were tests that I had to pull out of earlier class because the assert statements were more closely tied to the representation. Rather than doing if/then inside earlier tests, I put them separate here.

The problem was that I struggled to come up with a good name :laughing: